command/command.go

```diff
@@ -82,34 +100,30 @@ func (c *Command) Update() (updated bool, err error) {
 		updated = true
 	}
 
-	c.logger().Debugf("updated: %v - oldVersion: %q - newVersion: %q", updated, oldVersion, newVersion)
-
 	return updated, nil
 }
 
-// Status check the command's process state and returns
-// a verbose status.
-func (c *Command) Status() string {
+// Status check the command's process state and returns a verbose status.
+func (c *Command) Status() Status {
 	if c.Cmd.ProcessState == nil {
 		return c.status
 	}
 
 	ps := c.Cmd.ProcessState
```

Comment:
Yes, the truth is I didn't know how to get the correct status, and after debugging and fixing the code it ended up like that, I didn't even try to refactor it, but it always should look easier, like your approach.